tests/Keboola/ConfigMigrationTool/Migrations/OAuthMigrationTest.php

```diff
@@ -0,0 +1,145 @@
+        $this->oauthCredentials[] = $this->createOAuthConfigCustom($this->configurations[2]);
+        $this->oauthCredentials[] = $this->createOAuthConfigCustom($this->configurations[3]);
+    }
+
+    protected function createTestConfiguration(string $componentId) : string
+    {
+        $id = uniqid('migration-test');
+
+        $c = new Configuration();
+        $c->setComponentId($componentId);
+        $c->setConfigurationId($id);
+        $c->setName($id);
+        $c->setDescription('Migrate this account');
+        $c->setConfiguration(['authorization' => ['oauth_api' => ['id' => $id]]]);
+        $this->components->addConfiguration($c);
+
+        return $id;
+    }
+
+    protected function createOAuthConfig(string $configId) : \stdClass
+    {
+        return $this->oauthService->createCredentials($this->componentId, [
+            'id' => $configId,
+            'email' => 'test@keboola.com',
+            'accessToken' => '12345',
+            'refreshToken' => '567890',
+        ]);
+    }
+
+    protected function createOAuthConfigCustom(string $configId): \stdClass
+    {
+        return $this->oauthService->createCredentials($this->componentId, [
+            'id' => $configId,
+            'email' => 'test@keboola.com',
+            'accessToken' => '12345',
+            'refreshToken' => '567890',
+            'appKey' => 'appKey.12345.google.com',
+            'appSecretDocker' => 'KBC::ComponentSecure::tajmostvo',
+            'data' => [
+                'access_token' => 'qwertyuiop1234567890',
+            ],
+        ]);
+    }
+
+    protected function createMigrationConfig(string $configId) : array
+    {
+        return [
+            'componentId' => $this->componentId,
+            'id' => $configId,
+        ];
+    }
+
+    public function testExecute() : void
+    {
+        $responses = [];
+
+        foreach ($this->configurations as $index => $configurationId) {
+            $migration = new OAuthMigration(
+                $this->createMigrationConfig($configurationId),
+                new Logger(APP_NAME)
+            );
+            $response = $migration->execute();
+
+            $this->assertArrayHasKey('id', $response);
+            $this->assertArrayHasKey('authorizedFor', $response);
+            $this->assertArrayHasKey('creator', $response);
+            $this->assertArrayHasKey('created', $response);
+            $this->assertArrayHasKey('#data', $response);
+            $this->assertArrayHasKey('oauthVersion', $response);
+            $this->assertArrayHasKey('appKey', $response);
+            $this->assertArrayHasKey('#appSecret', $response);
+
+            $this->assertEquals('test@keboola.com', $response['authorizedFor']);
+            $this->assertEquals($configurationId, $response['id']);
+
+            $oldCredentials = $this->oauthCredentials[$index];
+            $this->assertEquals($oldCredentials->{'#data'}, $response['#data']);
+
+            if ($index > 1) {
```

Comment:
ja nejak nechapu proc je tu tahle podminka, ale jestli je tu schvalne, tak by to melo mit assert na count($this->configurations), protoze jinak se ten nasledujici assert vubec nemusi udelat

Reply:
Je to preto, ze prve dve konfiguracie v poli `$this->configurations` su obycajne a tie druhe dve su "custom". Viz riadok 53 v metode setUp sa to plni.

A potom dalej v tomto teste je `$this->assertNotEmpty($responses);` Aha, ale chapem ako to myslis, musi byt ten count vacsi ako dva. 

Tiez sa mi uplne nepaci ako to mam spravene. Mas navrh ako by to cele malo vyzerat?

Reply:
aha, tu inicializaci jsem nejak prehlidl, jo musi byt vetsi nez dva - nebo muze byt rovnou =4 :)